x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -422,6 +467,85 @@ protected void masterOperation(
         }));
     }
 
+    private boolean isForceMergeEnabled(Map<String, Object> sourceIndexMappings) {
+        if (sourceIndexMappings.containsKey("_meta")) {
+            if (sourceIndexMappings.get("_meta") instanceof Map<?, ?> metadataMap) {
+                var enabledForceMergeValue = metadataMap.get("downsample.forcemerge.enabled");
+                if (enabledForceMergeValue instanceof Boolean enabledForceMerge) {
+                    return enabledForceMerge;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Shortcircuit when another downsample api invocation already completed successfully.
+     */
+    private boolean canShortCircuit(
```

Comment:
Can the existing `canShortCircuit()` be reused? I think it is fine the pass down false for force merge in this case. The other downsample invocation should eventually do the force merge. So we don't need the variant that fetches mapping.

Reply:
Done, we discussed offline and we decided to default to true for force merge in order to preserve current behaviour.